components/cam/src/physics/cam/shoc.F90

```diff
@@ -1993,14 +1993,15 @@ subroutine shoc_assumed_pdf(&
   real(rtype) :: qw_sec_zt(shcol,nlev)
 
   ! define these so they don't have to be computed more than once
-  real(rtype), parameter :: sqrt2 = sqrt(2._rtype)
-  real(rtype), parameter :: sqrtpi = sqrt(2._rtype*3.14_rtype)
+  real(rtype) :: sqrt2, sqrtpi
+  sqrt2=bfb_sqrt(2._rtype)
+  sqrtpi=bfb_sqrt(2._rtype*3.14_rtype)
```

Comment:
@bogensch would it make sense in a separate PR to use pi to full precision here?

Reply:
Also, is this meant to be sqrt(pi) or is it really sqrt(2*pi)?

Reply:
It is meant to be sqrt(2*pi)... and yes, I agree that full precision pi should be used.

Reply:
I'll create one.

Reply:
Could we please use the name `sqrt2pi` instead of `sqrtpi` for `sqrt(2*pi)` ? Otherwise the name is misleading.

Reply:
In  `physics_constants.hpp`, pi is defined `pi=3.14159265`. Shouldn't it be more digits (in both places)? Should I add a constant to `shoc.F90` similarly?

Reply:
Since pi is a universal constant, I would put it in `physics/share` at the very least. It might be actually belong into a more common package (see E3SM-Project/EKAT#48 as well).

Either way: why forcing full precision? This is F90, so it probably doesn't matter, but if the same design stays in C++, isn't there the chance that it will cause all downstream calculation to be in double precision? It seems to me that, unless there's a clear _need_ for pi in double precision, then its precision should match that of the rest of the library...

Reply:
Here is a [PR for adding pi to shoc.F90](https://github.com/E3SM-Project/scream/pull/677) (instead of using 3.14) and changing the name to `sqrt2pi`.